incubator/hnc/pkg/validators/hierarchy.go

```diff
@@ -382,13 +382,16 @@ func deny(reason metav1.StatusReason, msg string) admission.Response {
 			},
 		}}
 	} else {
-		// metav1.StatusReasonInvalid shows the custom message in the Details field instead of
-		// Message field of metav1.Status.
+		// We need to set the custom message in both Message and Details fields.
+		// When interacting via the kutectl plugin, the custom message
+		// in the Message field will be shown when an error occurs. When interacting
+		// without the kubectl plugin, the custom message in the Details field will be shown.
```

Comment:
Eh? That's odd. Is this behaviour documented anywhere? Or is our plugin not treating the Details field properly?

Reply:
I don't think it is related to our plugin. When tracing back the call tree of [Error()](https://github.com/kubernetes-sigs/multi-tenancy/blob/f6a9f3e724d0ec5f9bbbc5a6e9351bc7a40719c8/incubator/hnc/pkg/kubectl/root.go#L150), it [calls ](https://github.com/kubernetes/client-go/blob/cb664d40f84c27bee45c193e4acb0fcd549b0305/rest/request.go#L1291) [FromObject](https://github.com/kubernetes/apimachinery/blob/7e441e0f246a2db6cf1855e4110892d1623a80cf/pkg/api/errors/errors.go#L100), which generates a [StatusError](https://github.com/kubernetes/apimachinery/blob/7e441e0f246a2db6cf1855e4110892d1623a80cf/pkg/api/errors/errors.go#L35) from `metav1.Status` (this is the type of [AdmissionResponse.Result](https://github.com/kubernetes/api/blob/464a0b5499224886bb4a6eeb41b33b79be9af1c8/admission/v1beta1/types.go#L127)). The `StatusError` implements the Error interface using [Message](https://github.com/kubernetes/apimachinery/blob/7e441e0f246a2db6cf1855e4110892d1623a80cf/pkg/api/errors/errors.go#L49) field from the status. IIUC, that's why when we print `Error()` it only prints the `Message` field.

Reply:
No, I don't think that's need, but can you please update this comment to include everything we've discussed? Because it is super-confusing. And don't try to write a short comment, make it as long and rambling as you like (e.g. https://github.com/kubernetes-sigs/multi-tenancy/blob/master/incubator/hnc/pkg/reconcilers/hierarchy_config.go#L306) - give any future reader as much of a chance as you can that they'll be able to figure out why we did this.

Reply:
Thanks for the suggestion! I added a detailed the comment here. PTAL. I also added code links in the comment, feel free to let me know if it makes the comment less readable or any suggestions to improve the readability. Thank you!

Reply:
Perfect :)